In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [3]:
x_train_data = pd.read_csv('dataset/x_train.csv')
y_train_data = pd.read_csv('dataset/y_train.csv')

In [13]:
x_test_data = pd.read_csv('dataset/x_test.csv')
y_test_data = pd.read_csv('dataset/y_test.csv')

# Linear Regression

In [5]:
from sklearn.linear_model import LinearRegression,Ridge

In [6]:
from sklearn.model_selection import learning_curve,cross_val_score,validation_curve

In [16]:
from sklearn.metrics import mean_squared_error,r2_score

In [18]:
from math import sqrt

In [8]:
linreg = LinearRegression()
linreg.fit(x_train_data, y_train_data)

LinearRegression()

In [9]:
LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)

LinearRegression(normalize=False)

In [11]:
y_predict = linreg.predict(x_test_data)

In [20]:
print(sqrt(mean_squared_error(y_test_data,y_predict)))
print((r2_score(y_test_data,y_predict)))


0.5874199592324111
0.6460551884502899


WIth linear regression: 
R2 = 0.65
RMSE = 58741.2 there's an average error of $58741.2 in prediction of house price